test/libsolidity/semanticTests/array/copying/copy_dynamically_sized_storage_array.sol

```diff
@@ -0,0 +1,36 @@
+contract C {
+    function testFunction1() public {}
+    function testFunction2() public {}
+    function testFunction3() public {}
+
+
+    function() external [] externalArray0 = new function() external[] (3);
+    function() external [] externalArray1 = [this.testFunction1, this.testFunction2, this
+    .testFunction3];
+
+    function copyExternalStorageArrayOfFunctionType() external returns (bool)
+    {
+        externalArray0 = externalArray1;
+        return keccak256(abi.encodePacked(externalArray0)) == keccak256(abi.encodePacked(externalArray1));
+    }
+
+    function() internal [] internalArray0 = new function() internal[] (3);
+    function() internal [] internalArray1 = [testFunction1, testFunction2, testFunction3];
+
+    function copyInternalArrayOfFunctionType() external returns (bool)
+    {
+        internalArray0 = internalArray1;
+        for(uint i = 0; i<internalArray0.length; i++)
+        {
+            if (internalArray0[i] != internalArray1[i])
+                return false;
+        }
+        return true;
```

Comment:
This will not detect the case where the arrays are different lengths but have equal initial elements :)

The array is short and has hard-coded length so it might be simpler to just compare elements manually here:
```suggestion
        assert(internalArray1.length == 3);
        return
            internalArray0.length == 3 &&
            internalArray0[0] == internalArray1[0] &&
            internalArray0[1] == internalArray1[1] &&
            internalArray0[2] == internalArray1[2];
```